detection_rules/packaging.py

```diff
@@ -320,7 +321,9 @@ def from_config(cls, config: dict = None, update_version_lock: bool = False, ver
         if log_deprecated:
             deprecated_rules = [r for r in all_rules if r.contents.metadata.maturity == 'deprecated']
 
-        rules = all_rules.filter(lambda r: filter_rule(r, rule_filter, exclude_fields))
+        package_version = Version(load_current_package_version())
+        rules = all_rules.filter(lambda r: (Version(r.contents.metadata.min_stack_version or '0')) <= package_version)
+        rules = rules.filter(lambda r: filter_rule(r, rule_filter, exclude_fields))
```

Comment:
Do we want to quietly drop them this way. Wouldn't we want to make sure they are never included in the branch?
```suggestion
```

Reply:
It sounds like you are referring to backporting, which the logic for exclusions with that would be separate from this. This would just ensure any packages built from main don't include "future" rules. Sound right?

Reply:
Right, so wouldn't a unit test be better?
Here, we just quietly drop rules, which could lead to a surprise. A unit test though should make that condition a hard failure

Reply:
This would be for more than just testing, but actually generating a "current" package.

Would it make you more comfortable if it was exposed as a function argument and then exposed in the `build-package` command?

Like:

```python
def from_config(cls, config: dict = None, update_version_lock: bool = False, max_stack_version: Optional[str] = None, verbose: bool = False) -> 'Package':
    max_stack_version = Version(max_stack_version or load_current_package_version())
```

Reply:
I just still don't understand the purpose.
Like we'll never have 7.14 rules in a 7.13 branch, or 7.15 rules in a 7.14 branch.

Here's another gotcha: let's say in the 7.14 branch, we make a a breaking change to a rule and update it's `min_stack_version` to 7.14.0. Then if within that branch, we say build a 7.13.0 filtered compatible package, it can be misleading because it won't be the same as the package generated from the 7.13 branch.

Reply:
I think I will remove it just to get this merged to not be a blocker to the other PR's, but I want to ensure we are both understanding this as well in order to address it later.

Think of this as only ever being used in main - it doesn't make sense in any release branch. 

Based on how it is written, only rules with a `min_stack_version` > `package_version` will ever validate against ECS/Beats master.

So if we are in main with a package release defined as `7.14`, that means the latest beats schema available would be `7.13.x`. If I know I am developing a rule to be compatible with an upcoming change, then I can label that rule `min_stack_version = "7.14.0"`, which would trigger it to validate against master schemas. In this same scenario, if I was in main and wanted to build a _current_ package, I would want to filter out _future_ rules.